TelemetryManager/Src/GroundStationCommunication.cpp

```diff
@@ -22,24 +22,14 @@ GroundStationCommunication::~GroundStationCommunication() {
 
 void GroundStationCommunication::transmit(TMCircularBuffer &transmissionBuffer) {
     // START: Send the bytes in transmissionBuffer to the ground station via RFD900
+    int bytesToTransmit = transmissionBuffer.bytesUntilLastMessageEnd();
+
+    for (int i {0}; i < bytesToTransmit; ++i) {
+        internalBuffer_[i] = transmissionBuffer.dequeue();
+    }
+    pRFD900->transmit(internalBuffer_, bytesToTransmit);
```

Comment:
Is there any logic to handle the case in which the bytes to transmit are longer than the RFD900_BUFF_SIZE? If not, are you able to brainstorm some ways of dealing with it? Thx.

Reply:
If they are longer, would it make sense in the context of mavlink for me to break up the msg and transmit the msg in bursts?

Reply:
There shouldn't be a situation in which a mavlink message is bigger than the transmit buffer. I was more thinking the situation in which we end up overwriting the length of the internal buffer because we have lets say 100 mavlink messages each ~200 bytes in length. For now, we can probably just stop dequeuing once the internal buffer is full. Then next time we transmit, just go from where we left off. basically we just need to check if the bytesToTransmit is longer than the internal buffer and if so, just set bytes to transmit to be the length of the buffer.